### ISSUE TITLE
About Mixup3D method

### ISSUE THREAD

**Issue by WYFDUT** [Traffic-X/MonoLSS#5]
Thank you very much for your excellent work!

In the mixup3d code, I found the following code somewhat illogical:

![图片2](https://github.com/Traffic-X/MonoLSS/assets/110314562/7bae27a3-374b-4fae-89c0-4adf4a028962)

This code, underlined in red, doesn't appear to take into account the distinction between the validation set and the training set. It is logical to take an image from self.idx_list list randomly (first take a random number whose maximum value is the length of the list, then use the list index to the corresponding value int(self.idx_list[random_num])). Here, random_num is directly used as an index to load images, and it is very likely to load images that do not belong to the training set but belong to the validation set.

Looking forward to your replies!

**Comment by jjrCN** [comment]
Thank you for your attention. For test data, this problem does not exist. What we finally uploaded only considered the completeness of the test set. For the validation set, this part needs to be changed to sampling from self.idx_list. Sorry for our negligence, we will modify this issue recently.

**Comment by WYFDUT** [comment]
> Thank you for your attention. For test data, this problem does not exist. What we finally uploaded only considered the completeness of the test set. For the validation set, this part needs to be changed to sampling from self.idx_list. Sorry for our negligence, we will modify this issue recently.

Thank you for your reply.

I would like to know if the above code modification will have any impact on the KITTI validation set metrics given in the paper.

If it does have an impact, would it be convenient to provide a copy of the results on the validation set after the code is correct?

**Comment by jjrCN** [comment]
This bug will have an impact on the validation set results. We are very sorry and thank you again for your help. We will update the arXiv article with new validation set results recently.

**Comment by WYFDUT** [comment]
> Thank you for your reply.

Thank you for your reply！Looking forward to your latest experimental data.

**Comment by jjrCN** [comment]
The arXiv paper version (https://arxiv.org/abs/2312.14474) has been updated. The new validation results are as follow. Thank you again for your attention and help.
<img width="607" alt="截屏2024-05-24 下午1 27 45" src="https://github.com/Traffic-X/MonoLSS/assets/37108754/9e06f616-2f68-47ed-94ab-744da3e35175">

**Comment by WYFDUT** [comment]
> The arXiv paper version (https://arxiv.org/abs/2312.14474) has been updated. The new validation results are as follow. Thank you again for your attention and help. <img alt="截屏2024-05-24 下午1 27 45" width="607" src="https://private-user-images.githubusercontent.com/37108754/333462532-9e06f616-2f68-47ed-94ab-744da3e35175.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3MjU1MTgsIm5iZiI6MTcxNjcyNTIxOCwicGF0aCI6Ii8zNzEwODc1NC8zMzM0NjI1MzItOWUwNmY2MTYtMmY2OC00N2VkLTk0YWItNzQ0ZGEzZTM1MTc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI2VDEyMDY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkNGMwMGE0NTQ0YTAzZDE4YWVhMjEyMjA3MzgyZWYxNDFjMzY4MWM1N2Y3NTAxNGE5MGY2ODVjMzNjOTg5YWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZMyxY52nl3zeTuK5sCv5hkUHzSFtfU0zH3gUZLVjDuc">

Thank you very much for your prompt reply with the new experimental data.

**Comment by johannes-tum** [comment]
Could this have also happened for the Waymo experiments?
